chombot-kcc/src/slash_commands/fancy_text.rs

```diff
@@ -0,0 +1,97 @@
+use std::error::Error;
+use std::fmt::{Display, Formatter};
+
+use anyhow::Result;
+use chombot_common::scraping_utils::{
+    create_chombot_http_client, create_chombot_http_client_insecure,
+};
+use chombot_common::{ChombotPoiseContext, ChombotPoiseUserData};
+use poise::serenity_prelude::{CreateAttachment, CreateMessage};
+use poise::CreateReply;
+
+#[derive(Debug)]
+pub enum FancyTextFetchError {
+    FetchError(anyhow::Error),
+    ParseError(anyhow::Error),
+}
+
+impl Display for FancyTextFetchError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::FetchError(err) => {
+                write!(f, "Could not fetch the fancy text response: {err}")
+            }
+            Self::ParseError(err) => {
+                write!(f, "Could not parse the fancy text response: {err}")
+            }
+        }
+    }
+}
+
+impl Error for FancyTextFetchError {}
+
+#[derive(Debug, Clone, serde::Deserialize)]
+struct FancyTextResponse {
+    #[serde(rename = "renderLocation")]
+    pub render_location: String,
+}
+
+const FANCY_TEXT_URL: &str = "https://cooltext.com/PostChange";
+
+/// Generate a fancy animated text.
+#[poise::command(slash_command)]
+pub async fn fancy_text<T: ChombotPoiseUserData>(
+    ctx: ChombotPoiseContext<'_, T>,
+    #[description = "The text to render"]
+    #[max_length = 100]
+    text: String,
+) -> Result<()> {
+    let client = create_chombot_http_client().map_err(FancyTextFetchError::FetchError)?;
+
+    let handle = ctx.say("Generating...").await?;
+
+    let body: FancyTextResponse = client
+        .post(FANCY_TEXT_URL)
+        .header("Accept", "*/*")
+        .header(
+            "Content-Type",
+            "application/x-www-form-urlencoded; charset=UTF-8",
+        )
+        .form(&[
+            ("LogoID", "4"),
+            ("Boolean1", "on"),
+            ("Integer13", "on"),
+            ("Text", &text),
+        ])
+        .send()
+        .await
+        .map_err(|err| FancyTextFetchError::FetchError(err.into()))?
+        .json()
+        .await
+        .map_err(|err| FancyTextFetchError::ParseError(err.into()))?;
+
+    let image = create_chombot_http_client_insecure()
```

Comment:
This is unacceptable; there shouldn't be any case of ignoring certificate validation for connections over the public Internet.

Reply:
It doesn't matter. You need to use different upstream of figure this out.

Reply:
We should strive to maintain high security standards and not push code like that.

Reply:
Sadly, it seems the upstream server is the root of the issue, as I can see the same certificate error across different operating systems and tooling. Also, I don't really see much of a danger here, since this is a static image anyway, and there's no private info being sent here, so TLS doesn't add much of a seqre-ity here.

Reply:
lmao